hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -192,7 +192,7 @@ public SizeEstimator<BufferedRecord<T>> getRecordSizeEstimator() {
   }
 
   public CustomSerializer<BufferedRecord<T>> getRecordSerializer() {
-    return new DefaultSerializer<>();
+    return new BufferedRecordSerializer<>();
```

Comment:
why this change? It seems not necessary based on the benchmark: https://github.com/apache/hudi/pull/13408#discussion_r2139076809